app/controllers/outputs_controller.rb

```diff
@@ -0,0 +1,97 @@
+class OutputsController < ApplicationController
+  ALLOWED_RESOURCE_TYPES = %w(study_impact dissemination publication)
+
+  def create
+    @study = Study.find(params[:study_id])
+    resource_type = params[:output_type]
+    if ALLOWED_RESOURCE_TYPES.include? resource_type
+      self.send("create_#{resource_type}")
+    else
+      raise ActionController::RoutingError.new('Not Found')
+    end
+  end
+
+  private
+
+  def create_study_impact
+    # The study impact form allows you to create multiple impacts via a
+    # multiple select checkbox, so we take an all-or-nothing approach to
+    # creating them
+    @study_impacts = {}
+    StudyImpact.transaction do
+      impact_type_ids = study_impact_params[:impact_type_ids]
+      descriptions = study_impact_params[:descriptions]
+      impact_type_ids.each do |key, impact_type_id|
```

Comment:
Unused block argument - `key`. If it's necessary, use `_` or `_key` as an argument name to indicate that it won't be used.